shopfloor_reception/services/reception.py

```diff
@@ -1398,6 +1400,61 @@ def set_destination(
             return response
         return self._response_for_select_move(picking)
 
+    def set_storage_type(self, picking_id, selected_line_id, barcode=""):
+        """Set a storage type on the result package."""
+        picking = self.env["stock.picking"].browse(picking_id)
+        selected_line = self.env["stock.move.line"].browse(selected_line_id)
+        message = self._check_picking_processible(picking)
+        if message:
+            return self._response_for_select_dest_package(
+                picking, selected_line, message=message
+            )
+        if not selected_line.exists():
+            message = self.msg_store.record_not_found()
+            return self._response_for_select_dest_package(
+                picking, selected_line, message=message
+            )
+        if barcode:
+            storage_type = self.env["stock.package.type"].search(
+                [("barcode", "=", barcode)]
+            )
+            message = self._check_storage_type_valid(storage_type)
+            if not message:
+                # FIXME : this is to avoid an exception when it is required
+                selected_line.result_package_id.height = 1
+                selected_line.result_package_id.package_type_id = storage_type
+                if hasattr(selected_line, "_recompute_putaways"):
+                    # Recompute the putaway location if the module
+                    # stock_picking_putaway_recompute is installed
+                    selected_line._recompute_putaways()
```

Comment:
Although I prefer glue modules, there are cases where this lazy dependency is useful. I lack context to know if this is the case.

But if you go for it, I recommend this pattern instead:

https://github.com/odoo/odoo/blob/2c02f5956f5db3137875d200133779740e19d724/addons/partner_autocomplete/models/res_partner.py#L186

```suggestion
                if self.env['ir.module.module']._get('stock_picking_putaway_recompute').state == 'installed':
                    # Recompute the putaway location if the module
                    # stock_picking_putaway_recompute is installed
                    selected_line._recompute_putaways()
```

This way, in case the `stock_picking_putaway_recompute` is refactored and somehow gets its `_recompute_putaways` method renamed or removed, you'd get a real execution error and it'd not be silently ignored.

Reply:
I agree, refactored. Although it was decided not to use a glue module for adding this feature.